src/NelsonSiegelSvensson.jl

```diff
@@ -0,0 +1,181 @@
+abstract type ParametricModel end
+
+"""
+    NelsonSiegel(β₀, β₁, β₂, τ₁)
+
+Parameters of Nelson and Siegel (1987) parametric model. 
+
+# Examples
+
+```julia-repl
+julia> β₀, β₁, β₂, τ₁ = 0.6, -1.2, -1.9, 3.0
+julia> nsm = NelsonSiegelSvensson.NelsonSiegel.(β₀, β₁, β₂, τ₁)
+```
+"""
+struct NelsonSiegel <: ParametricModel
+    β₀
+    β₁
+    β₂
+    τ₁
+    #r² no need
+
+    #function NelsonSiegel(β₀, β₁, β₂, τ₁, r²)
+    function NelsonSiegel(β₀, β₁, β₂, τ₁)
+        #(τ₁ <= 0 || r² < 0 || r² > 1) && throw(DomainError("Wrong parameter ranges"))
+        (τ₁ <= 0) && throw(DomainError("Wrong parameter ranges"))
+        #return new(β₀, β₁, β₂, τ₁, r²)
+        return new(β₀, β₁, β₂, τ₁)
+    end
+end
+
+"""
+    NelsonSiegelSvensson(β₀, β₁, β₂, β₃, τ₁, τ₂)
+
+Parameters of Svensson (1994) parametric model. 
+
+# Examples
+
+```julia-repl
+julia> β₀, β₁, β₂, β₃, τ₁, τ₂ = 0.6, -1.2, -2.1, 3.0, 1.5
+julia> nssm = NelsonSiegelSvensson.NelsonSiegelSvensson.(β₀, β₁, β₂, β₃, τ₁, τ₂)
+```
+"""
+struct NelsonSiegelSvensson <: ParametricModel
+    β₀
+    β₁
+    β₂
+    β₃
+    τ₁
+    τ₂
+    #r² no need
```

Comment:
Can the commented lines with this (misc lines) be removed then? How come it's not needed?

Reply:
They are commented out from one of earlier comments from @kasperrisager . They are kept I think until the code is finalized.

Reply:
As we are closer to merging, go ahead and delete them in a new commit - we can always look at prior commits for old code.